lib/src/global_packages.dart

```diff
@@ -90,9 +90,14 @@ class GlobalPackages {
     // dependencies. Their executables shouldn't be cached, and there should
     // be a mechanism for redoing dependency resolution if a path pubspec has
     // changed (see also issue 20499).
+    PackageRef ref;
+    try {
+      ref = cache.git.source.parseRef(name, {'url': repo}, containingPath: '.');
```

Comment:
Technically, we're going to write the `pubspec.yaml` somewhere other than `.`, so the `containingPath` should be something like `$PUB_CACHE/global_packages/<package>/`, shouldn't it?

I'm not sure it matters here?

Reply:
We don't write it to disk (yet), so it should not matter...